todo/src/repository/file_storage.rs

```diff
@@ -0,0 +1,63 @@
+use async_trait::async_trait;
+use tokio::fs::{read_to_string, write};
+pub(crate) mod error;
+use crate::domain::todo::Todo;
+use crate::domain::todos::Todos;
+use crate::repository::file_storage::error::StorageError;
+use std::path::PathBuf;
+
+pub struct FileStorage {
+    pub path: PathBuf,
+}
+
+#[async_trait]
+pub trait Storage {
+    async fn get_todos_from_filestorage(&self) -> Result<Todos, StorageError>;
+    async fn write_filestorage(&self, todo_list: &Todos) -> Result<(), StorageError>;
+}
+
+#[async_trait]
+impl Storage for FileStorage {
+    async fn get_todos_from_filestorage(&self) -> Result<Todos, StorageError> {
+        let todo_str = read_to_string(&self.path)
+            .await
+            .map_err(|_| StorageError::Read)?;
+
+        let vec_todo = todo_str
+            .lines()
+            .map(|line| FileStorage::build_todo(line))
+            .collect::<Result<Vec<Todo>, _>>()?;
+
+        Ok(Todos::new(vec_todo))
+    }
+
+    async fn write_filestorage(&self, todo_list: &Todos) -> Result<(), StorageError> {
+        let mut todo_list_str = String::new();
+
+        for todo in todo_list.iter() {
+            let item_list = match todo.done {
+                true => format!("[X] - {}\n", &todo.message.to_string()),
+                false => format!("[ ] - {}\n", &todo.message.to_string()),
```

Comment:
Aqui você _poderia_ usar a implementação de `Display` do `Todo`, mas como a forma como você salva no arquivo é uma "preocupação" do seu Storage, acho legal deixar inline mesmo.

Só tirar os `.to_string()`s.
```suggestion
                true => format!("[X] - {}\n", &todo.message),
                false => format!("[ ] - {}\n", &todo.message),
```